File: src/main/webapp/outRoom.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
// handle when user want to quit room
$(document).ready(function(){
    
    $('#quitRoom').click(function(){
        // use ajax to send command to servlet, check if this user is the room's admin or not
        $.ajax({
            type: 'post',
            data: {command: 'quit'},
            url: 'OutRoom',
            success: function(result)
            {
                // if user is admin, couldn't quit
                if (result == 'false') alert("You are this Room's Admin. You are allowed to quit unless delete the room!");
                // if user is not admin, cound quit
                // room-name in chat-list will disappear
                else 
                {
                    $('#room-' + result).css("display", "none");
                    $('#insert-roomName1').html("");
                    // call function removeSocket on startChat.js
                    $('#removeSocket').click();
                    var event1 = document.getElementById("p1");
                    
                    event1.classList.toggle('col-9');
                    event1.classList.toggle('col-6');
                }
            }
        });
    });
    
    
    // handle if user want to delete this room
    $('#deleteRoom').click(function(){
        $.ajax({
            type: 'post',
            data: {command: 'delete'},
            url: 'OutRoom',
            success: function(result)
            {
                // if user is not admin of the room, can't delete
                if (result == 'false') alert("You are  not this Room's Admin. Can't delete the conversation!");
                else 
                {
                    $('#room-' + result).css("display", "none");
                    $('#insert-roomName1').html("");
                    $('#removeSocket').click();
                    var event1 = document.getElementById("p1");
                    
                    event1.classList.toggle('col-9');
                    event1.classList.toggle('col-6');
                }
            }
        });
    });

});

File: src/main/java/com/webchatproject/room/OutRoom.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.webchatproject.room;

import Model.User;
import com.webchatproject.connectdatabase.ConnectDatabase;
import java.io.IOException;
import java.io.PrintWriter;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.logging.Level;
import java.util.logging.Logger;
import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import javax.servlet.http.HttpSession;

/**
 *
 * @author HANG.VTT183524
 */
// this class handle when a user want to quit from a chat-room or admin want to delete a conversation
public class OutRoom extends HttpServlet {

    @Override
    protected void doGet(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        
    }

    @Override
    protected void doPost(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        response.setContentType("text/plain");
        
        String command = (String) request.getParameter("command");
        
        if (command.equals("quit")) quitRoom(request, response);
        if (command.equals("delete")) deleteRoom(request, response);
    }
   
    private void quitRoom(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException
    {
        HttpSession session = request.getSession();
        PrintWriter out = response.getWriter();
        
        User user = (User) session.getAttribute("user");
        int user_id = user.getUserId();
        String room_id = (String) session.getAttribute("room_id");
        
        // check is user who wanted to quit is admin of this room or not. 
        // if true, don't allow quit. Because only admin could delete a room. If admin quit before delete room, that room will be never deleted by someone
        ConnectDatabase connect = new ConnectDatabase();
        ResultSet rs = connect.executeSql("select creator_id from Chat_Room where room_id = " + room_id + " ;");
        
        try {
            if (rs.next())
            {
                if (user_id == Integer.parseInt(rs.getString("creator_id")))
                    out.print(false);
                else 
            {
                // if this user isn't admin, delete him from DB
                connect.insertIntoDatabase("delete from Participant where room_id = " + room_id + " and member_id = " + user_id + " ;");
                out.print(room_id);
                
            }
            }
            
            rs.close();
        } catch (SQLException ex) {
            Logger.getLogger(OutRoom.class.getName()).log(Level.SEVERE, null, ex);
        }
        connect.closeConnect();
    }
    
    private void deleteRoom(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException
    {
        HttpSession session = request.getSession();
        PrintWriter out = response.getWriter();
        
        User user = (User) session.getAttribute("user");
        int user_id = user.getUserId();
        String room_id = (String) session.getAttribute("room_id");
        
        ConnectDatabase connect = new ConnectDatabase();
        ResultSet rs = connect.executeSql("select creator_id from Chat_Room where room_id = " + room_id + " ;");
        
        // delete if user who click delete-button is admin of this room
        try {
            if (rs.next())
            {
                if (user_id == Integer.parseInt(rs.getString("creator_id")))
                {
                    connect.insertIntoDatabase("delete from Participant where room_id = " + room_id + " ;");
                    connect.insertIntoDatabase("delete from Messenger where room_id = " + room_id + " ;");
                    connect.insertIntoDatabase("delete from Chat_Room where room_id = " + room_id + " ;");
                    out.print(room_id);
                }
                else out.print(false);
            }
            rs.close();
        } catch (SQLException ex) {
            Logger.getLogger(OutRoom.class.getName()).log(Level.SEVERE, null, ex);
        }
        connect.closeConnect();
    }

    /**
     * Returns a short description of the servlet.
     *
     * @return a String containing servlet description
     */
    @Override
    public String getServletInfo() {
        return "Short description";
    }// </editor-fold>

}
